sources/functions/users

```diff
@@ -108,22 +108,27 @@ function password_check() {
 
     #Perform tests. If failure is detected, recurse by unsetting $pass, and running `password_check $ui`
     ##Cracklib test
-    echo_progress_start "Checking password strength"
-    result=$(cracklib-check <<< "$pass")
-    if ! grep -q OK <<< "$result"; then
-        result=""
-        unset result
-        if [[ $ui = "whiptail" ]]; then
-            whiptail --title "Error" --msgbox "Password did not pass cracklib check." --ok-button="Try again" 10 50
+    if [ "$SKIPCRACKLIB" != "true" ]; then
+        echo_progress_start "Checking password strength"
+        result=$(cracklib-check <<< "$pass")
+        if ! grep -q OK <<< "$result"; then
+            crackreason=$(echo "$result" | cut -d':' -f2)
+            if [[ $ui = "whiptail" ]]; then
+                whiptail --title "Error" --msgbox "Password did not pass cracklib check.\nReason: \"${crackreason/ /}\"" --ok-button="Try again" 10 50
+            else
+                echo_warn "Password did not pass cracklib check. Reason: \"${crackreason/ /}\". Press enter to try again"
+                read -n 1 -r
+            fi
+            result=""
+            unset result
+            pass=
+            unset pass
+            password_check "$1"
         else
-            echo_warn "Password did not pass cracklib check. Press enter to try again"
-            read -n 1 -r
+            echo_progress_done "Password passed"
         fi
-        pass=
-        unset pass
-        password_check "$1"
-    else
-        echo_progress_done "Password passed"
+        SKIPCRACKLIB=""
+        unset SKIPCRACKLIB
```

Comment:
Have you tested this? In my experience, once cracklib is installed passwords will start failing to set if they don't meet cracklib requirements. Meaning cracklib cannot be installed at all, not just bypassed

Reply:
Hmmm, seems to work with a generic "password" password under Buster